audiobook_generator/book_parsers/epub_book_parser.py

```diff
@@ -71,9 +65,29 @@ def get_chapters(self, break_string) -> List[Tuple[str, str]]:
                 cleaned_text = re.sub(r'(?<=[a-zA-Z.,!?;”")])\d+', "", cleaned_text)
                 logger.debug(f"Cleaned text step 4: <{cleaned_text[:100]}>")
 
-            # fill in the title if it's missing
-            if title == "":
+            # Get proper chapter title
+            if self.config.title_mode == "auto":
+                title = ""
+                title_levels = ['title', 'h1', 'h2', 'h3']
+                for level in title_levels:
+                    if soup.find(level):
+                        title = soup.find(level).text
+                        break
+                if title == "" or re.match(r'^\d{1,3}$',title) is not None:
```

Comment:
When the extracted title is something like `1`, `12`, `123`, the first few texts in that chapter will be used instead. LGTM.